tests/test_client.py

```diff
@@ -98,6 +99,10 @@ def test_invalid_url(self) -> None:
         with pytest.raises(TypeError):
             Client.open()  # type: ignore[call-arg]
 
+    def test_client_open_timeout(self) -> None:
+        with pytest.raises(APIError):
+            Client.open("http://10.255.255.1", timeout=1)
+
```

Comment:
This isn't working yet.  Evidently, the APIError observed when calling this manually isn't the error that needs to be caught.  Could use a pointer here.

Reply:
I'd suggest using a [**requests-mock** dynamic response](https://requests-mock.readthedocs.io/en/latest/response.html#dynamic-response) and a [sleep](https://docs.python.org/3/library/time.html#time.sleep) to build your own slow-responding response. That way you don't have to rely on `10.255.255.1`'s behavior.

Reply:
Tried mock, but that false function is not triggering the required timeout behavior.  Thoughts?